tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -266,11 +266,28 @@ void CustomActionData::ensureDomainHasCorrectFormat()
             WcaLog(LOGMSG_STANDARD, "Supplied domain name \"%S\"", _user.Domain.c_str());
             _domainUser = true;
         }
-        else if (_wcsicmp(_user.Domain.c_str(), L"NT AUTHORITY") != 0) // NT Authority should never be considered a "domain".
+        else
         {
-            WcaLog(LOGMSG_STANDARD, "Warning: Supplied user in different domain (\"%S\" != \"%S\")", _user.Domain.c_str(),
-                   _targetMachine->DnsDomainName().c_str());
-            _domainUser = true;
+            // Compute a temporary fully qualified username to retrieve its SID
+            // in order to determine if its prefix starts with NT AUTHORITY.
+            auto tempFquname = _user.Domain + L"\\" + _user.Name;
+            auto sidResult = GetSidForUser(nullptr, tempFquname.c_str());
+            if (sidResult.Result != ERROR_NONE_MAPPED)
+            {
+                const auto ntAuthoritySid = WellKnownSID::NTAuthority();
+                if (!ntAuthoritySid.has_value())
+                {
+                    WcaLog(LOGMSG_STANDARD, "Cannot check user SID against NT AUTHORITY: memory allocation failed");
+                }
+                else if (!EqualPrefixSid(
+                             sidResult.Sid.get(),
+                             ntAuthoritySid.value().get())) // NT Authority should never be considered a "domain".
+                {
+                    WcaLog(LOGMSG_STANDARD, "Warning: Supplied user in different domain (\"%S\" != \"%S\")",
+                           _user.Domain.c_str(), _targetMachine->DnsDomainName().c_str());
```

Comment:
I wonder why it is classified as a warning. It is Ok IMHO to specify particular domain name during installation.

Reply:
IMHO it is better but it was there before your change. So it is totally up to you.